### PR TITLE
python: add SR-IOV support

### PR DESCRIPTION
**Pull request by ffmancera**
Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

File: src/python/nispor/iface.py [hunk 1]
```diff
@@ -55,6 +56,8 @@ def _iface_info_to_obj(iface_info):
         iface = NisporBond(iface_info)
     elif iface_type == "Bridge":
         iface = NisporBridge(iface_info)
+    elif iface_type == "Ethernet":
+        iface = NisporSriov(iface_info)
```
Comment (cathay4t) [inline src/python/nispor/iface.py]: Please only use `NisporSriov` only when SR-IOV entry found.

Calling all ethenrnet interface as `NisporSriov` is not a good idea.

File: src/python/nispor/sriov.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @property
+    def number_vfs(self):
+        return len(self._sriov_info.get("vfs", []))
+
+    @property
+    def vfs(self):
+        return [NisporSriovVf(vf) for vf in self._sriov_info.get("vfs", [])]
+
+
+class NisporSriovVf(NisporSriov):
```
Comment (cathay4t) [inline src/python/nispor/sriov.py]: `NisporSriovVf` is not a child class of `NisporSriov`

File: src/python/nispor/sriov.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+class NisporSriov(NisporBaseIface):
+    def __init__(self, info):
+        super().__init__(info)
+        self._sriov_info = self._info.get("sr-iov", {})
+
+    @property
+    def number_vfs(self):
+        return len(self._sriov_info.get("vfs", []))
```
Comment (cathay4t) [inline src/python/nispor/sriov.py]: There is no need to expose this property.

**Review by cathay4t — CHANGES_REQUESTED** [review]
You cannot simple set all properties as optional.

According to `VfInfo` in rust, many of them are mandatory.

File: src/python/nispor/iface.py (hunk 1 above)
Comment (ffmancera) [reply src/python/nispor/iface.py]: Done, thanks!

**Comment by cathay4t** [comment]
I believe we don't need this PR. The SR-IOV info is stored as `NisporBaseIface.sriov` property.
Sorry for wasting your time.

**Pull request closed without merging**